Script-triggered hit testing should use `RenderStyle::visibility()` instead of `usedVisibility`

whsieh opened the pull request:
#### f8bb1dc584498cbba4b6609c3e0ec6a115996ce2
<pre>
Script-triggered hit testing should use `RenderStyle::visibility()` instead of `usedVisibility`
<a href="https://bugs.webkit.org/show_bug.cgi?id=271413">https://bugs.webkit.org/show_bug.cgi?id=271413</a>

Reviewed by Antti Koivisto and Anne van Kesteren.

Introduce a `HitTestSource` enum which indicates whether or not hit-testing was triggered by (or is
otherwise directly observable by) script running on the webpage, and plumb it through the following
programmatic hit-testing APIs:

• `DocumentOrShadowRoot.elementFromPoint()`
• `DocumentOrShadowRoot.elementsFromPoint()`
• `Document.caretRangeFromPoint()`

This allows programmatically-driven hit testing to ignore `usedVisibility()` styles, while allowing
user-triggered (or native-API-triggered) hit-testing to honor the used visibility. In short, the
presence of `HitTestSource::Script` means that we:

1.  Avoid honoring `usedVisibility()` when getting styles on `RenderStyle` directly, or...
2.  Ignore `usedVisibility()` when performing a hit-test that&apos;s triggered by script.

See below for more details.

* LayoutTests/fast/element-targeting/basic-element-targeting.html:

Augment an existing layout test to exercise this change.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::characterOffsetForPoint):
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::visiblePositionRangeForLine const):
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::visiblePositionForPoint const):

Explicitly pass in `HitTestSource::User` for all of these accessibility-related call sites.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::caretRangeFromPoint):
(WebCore::Document::caretPositionFromPoint):
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/Document.idl:
* Source/WebCore/dom/DocumentOrShadowRoot.idl:
* Source/WebCore/dom/TreeScope.cpp:
(WebCore::TreeScope::nodeFromPoint):
(WebCore::TreeScope::elementFromPoint):
(WebCore::TreeScope::elementsFromPoint):
* Source/WebCore/dom/TreeScope.h:

Introduce a `HitTestSource` argument for these three DOM APIs above, and plumb the source over to
the underlying `HitTestRequest`.

* Source/WebCore/editing/Editing.cpp:
(WebCore::closestEditablePositionInElementForAbsolutePoint):

Pass in `User` here, since this helper method is only used for user-triggered actions.

* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::FrameSelection::contains const):
* Source/WebCore/editing/VisibleUnits.cpp:
(WebCore::previousLinePosition):
(WebCore::nextLinePosition):

Default to `HitTestSource::Script` for now, since these editing codepaths can be directly triggered
from script. In a future patch, we&apos;ll need to plumb this state down from bindings calls that trigger
this editing code — for instance, `Document.execCommand` and `DOMSelection.modify`.

* Source/WebCore/editing/cocoa/DataDetection.mm:
(WebCore::DataDetection::detectItemAroundHitTestResult):
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayBox.h:
(WebCore::InlineDisplay::Box::isVisibleIgnoringUsedVisibility const):
* Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp:
(WebCore::LayoutIntegration::LineLayout::hitTest):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::selectClosestWordFromHitTestResult):
(WebCore::EventHandler::selectClosestContextualWordFromHitTestResult):
(WebCore::EventHandler::selectClosestContextualWordOrLinkFromHitTestResult):
(WebCore::EventHandler::handleMousePressEventTripleClick):
(WebCore::EventHandler::handleMousePressEventSingleClick):
(WebCore::EventHandler::selectionExtentRespectingEditingBoundary):
(WebCore::EventHandler::updateSelectionForMouseDrag):
(WebCore::EventHandler::handleMouseReleaseEvent):

Pass in `User` in more places.

* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::visiblePositionForPoint const):
* Source/WebCore/rendering/HitTestRequest.h:

Add a new `HitTestSource` member to `HitTestRequest` that indicates whether the hit test request is
directly invoked through `Script`, or if it&apos;s user-triggered. We use this flag to prefer specified
`visibility()` over `usedVisibility()` (effectively ignoring all visibility adjustment).

(WebCore::HitTestRequest::assertConsistentType):
(WebCore::HitTestRequest::HitTestRequest):
(WebCore::HitTestRequest::userTriggered const):
* Source/WebCore/rendering/HitTestSource.h: Added.
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::positionForPointRespectingEditingBoundaries):
(WebCore::RenderBlock::positionForPointWithInlineChildren):
(WebCore::isChildHitTestCandidate):
(WebCore::RenderBlock::positionForPoint):
* Source/WebCore/rendering/RenderBlock.h:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::positionForPointWithInlineChildren):
(WebCore::RenderBlockFlow::positionForPoint):
* Source/WebCore/rendering/RenderBlockFlow.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::positionForPoint):
* Source/WebCore/rendering/RenderBox.h:
* Source/WebCore/rendering/RenderElementInlines.h:
(WebCore::RenderElement::visibleToHitTesting const):
* Source/WebCore/rendering/RenderFileUploadControl.cpp:
(WebCore::RenderFileUploadControl::positionForPoint):
* Source/WebCore/rendering/RenderFileUploadControl.h:
* Source/WebCore/rendering/RenderFragmentContainer.cpp:
(WebCore::RenderFragmentContainer::positionForPoint):
* Source/WebCore/rendering/RenderFragmentContainer.h:
* Source/WebCore/rendering/RenderInline.cpp:
(WebCore::RenderInline::positionForPoint):
* Source/WebCore/rendering/RenderInline.h:
* Source/WebCore/rendering/RenderLineBreak.cpp:
(WebCore::RenderLineBreak::positionForPoint):
* Source/WebCore/rendering/RenderLineBreak.h:
* Source/WebCore/rendering/RenderMultiColumnSet.cpp:
(WebCore::RenderMultiColumnSet::positionForPoint):
* Source/WebCore/rendering/RenderMultiColumnSet.h:
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::positionForPoint):
* Source/WebCore/rendering/RenderObject.h:

Plumb this new `HitTestRequest` flag through `RenderObject::positionForPoint()`, and propagate it
through all call sites and subclass implementations, recursively.

* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::RenderReplaced::paint):
* Source/WebCore/rendering/RenderReplaced.h:
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::positionForPoint):
* Source/WebCore/rendering/RenderText.h:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::isVisibleToHitTesting):
* Source/WebCore/rendering/svg/RenderSVGImage.cpp:
(WebCore::RenderSVGImage::nodeAtPoint):
* Source/WebCore/rendering/svg/RenderSVGInlineText.cpp:
(WebCore::RenderSVGInlineText::positionForPoint):
* Source/WebCore/rendering/svg/RenderSVGInlineText.h:
* Source/WebCore/rendering/svg/RenderSVGShape.cpp:
(WebCore::RenderSVGShape::nodeAtPoint):
* Source/WebCore/rendering/svg/RenderSVGText.cpp:
(WebCore::RenderSVGText::nodeAtFloatPoint):
(WebCore::RenderSVGText::nodeAtPoint):
(WebCore::RenderSVGText::positionForPoint):
* Source/WebCore/rendering/svg/RenderSVGText.h:
* Source/WebCore/rendering/svg/SVGInlineTextBox.cpp:
(WebCore::SVGInlineTextBox::nodeAtPoint):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGImage.cpp:
(WebCore::LegacyRenderSVGImage::nodeAtFloatPoint):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGShape.cpp:
(WebCore::LegacyRenderSVGShape::nodeAtFloatPoint):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::computeElementLayout):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDocumentGtk.cpp:
(webkit_dom_document_element_from_point):
(webkit_dom_document_caret_range_from_point):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::selectionPositionInformation):
* Source/WebKitLegacy/mac/DOM/DOMDocument.mm:
(-[DOMDocument elementFromPoint:y:]):
(-[DOMDocument caretRangeFromPoint:y:]):

Canonical link: <a href="https://commits.webkit.org/276762@main">https://commits.webkit.org/276762@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0a0c1d3cf9c96ced61006d567b31e24af8568184

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/45542 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/24668 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48081 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48212 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/41551 "Built successfully") 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/47849 "Build is in progress. Recent messages:OS: Unknown (10.15.7), Xcode: 11.7") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/28922 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22062 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37336 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46120 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/49/builds/28922 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39300 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18461 "Passed tests") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/49/builds/28922 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40388 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/3588 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-ios](https://ews-build.webkit.org/#/builders/49/builds/28922 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/40717 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/49948 "Build is in progress. Recent messages:Printed configuration") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/20530 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/51/builds/22062 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/49948 "Build is in progress. Recent messages:Printed configuration") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/21836 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/39452 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/49948 "Build is in progress. Recent messages:Printed configuration") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10132 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22196 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/21524 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/dom/Document.idl:
```diff
@@ -87,7 +87,7 @@ typedef (HTMLScriptElement or SVGScriptElement) HTMLOrSVGScriptElement;
 
     // Non standard: It has been superseeded by caretPositionFromPoint which we do not implement yet.
-    Range caretRangeFromPoint(optional long x = 0, optional long y = 0);
+    [ImplementedAs=caretRangeFromPointForBindings] Range caretRangeFromPoint(optional long x = 0, optional long y = 0);
```
I would prefer doing this the other way around here and elsewhere. At least I was taught that we generally aim for the C++ API to match the web standard API.

whsieh replied:
Got it — as we discussed over Slack, we should actually be able to remove these `-ForBindings` variants altogether, and instead just have `caretRangeFromPoint(int x, int y, HitTestSource = HitTestSource::Script)`. The internal call site can just pass in `HitTestSource::User` directly.

annevk commented on Source/WebCore/rendering/HitTestSource.h:
```diff
@@ -0,0 +1,35 @@
+namespace WebCore {
+
+enum class HitTestSource : bool {
+    Bindings,
```
Script/Website?

whsieh replied:
Sounds good to me! Will replace `HitTestSource::Bindings` -> `HitTestSource::Script`

annevk reviewed (approved):
This looks good to me. I have some optional nits inline. I'd feel more comfortable if a longer term reviewer would also take a look though.

whsieh committed:
Script-triggered hit testing should use `RenderStyle::visibility()` instead of `usedVisibility`
https://bugs.webkit.org/show_bug.cgi?id=271413

Reviewed by Antti Koivisto and Anne van Kesteren.

Introduce a `HitTestSource` enum which indicates whether or not hit-testing was triggered by (or is
otherwise directly observable by) script running on the webpage, and plumb it through the following
programmatic hit-testing APIs:

• `DocumentOrShadowRoot.elementFromPoint()`
• `DocumentOrShadowRoot.elementsFromPoint()`
• `Document.caretRangeFromPoint()`

This allows programmatically-driven hit testing to ignore `usedVisibility()` styles, while allowing
user-triggered (or native-API-triggered) hit-testing to honor the used visibility. In short, the
presence of `HitTestSource::Script` means that we:

1.  Avoid honoring `usedVisibility()` when getting styles on `RenderStyle` directly, or...
2.  Ignore `usedVisibility()` when performing a hit-test that's triggered by script.

See below for more details.

* LayoutTests/fast/element-targeting/basic-element-targeting.html:

Augment an existing layout test to exercise this change.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::characterOffsetForPoint):
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::visiblePositionRangeForLine const):
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::visiblePositionForPoint const):

Explicitly pass in `HitTestSource::User` for all of these accessibility-related call sites.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::caretRangeFromPoint):
(WebCore::Document::caretPositionFromPoint):
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/Document.idl:
* Source/WebCore/dom/DocumentOrShadowRoot.idl:
* Source/WebCore/dom/TreeScope.cpp:
(WebCore::TreeScope::nodeFromPoint):
(WebCore::TreeScope::elementFromPoint):
(WebCore::TreeScope::elementsFromPoint):
* Source/WebCore/dom/TreeScope.h:

Introduce a `HitTestSource` argument for these three DOM APIs above, and plumb the source over to
the underlying `HitTestRequest`.

* Source/WebCore/editing/Editing.cpp:
(WebCore::closestEditablePositionInElementForAbsolutePoint):

Pass in `User` here, since this helper method is only used for user-triggered actions.

* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::FrameSelection::contains const):
* Source/WebCore/editing/VisibleUnits.cpp:
(WebCore::previousLinePosition):
(WebCore::nextLinePosition):

Default to `HitTestSource::Script` for now, since these editing codepaths can be directly triggered
from script. In a future patch, we'll need to plumb this state down from bindings calls that trigger
this editing code — for instance, `Document.execCommand` and `DOMSelection.modify`.

* Source/WebCore/editing/cocoa/DataDetection.mm:
(WebCore::DataDetection::detectItemAroundHitTestResult):
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayBox.h:
(WebCore::InlineDisplay::Box::isVisibleIgnoringUsedVisibility const):
* Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp:
(WebCore::LayoutIntegration::LineLayout::hitTest):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::selectClosestWordFromHitTestResult):
(WebCore::EventHandler::selectClosestContextualWordFromHitTestResult):
(WebCore::EventHandler::selectClosestContextualWordOrLinkFromHitTestResult):
(WebCore::EventHandler::handleMousePressEventTripleClick):
(WebCore::EventHandler::handleMousePressEventSingleClick):
(WebCore::EventHandler::selectionExtentRespectingEditingBoundary):
(WebCore::EventHandler::updateSelectionForMouseDrag):
(WebCore::EventHandler::handleMouseReleaseEvent):

Pass in `User` in more places.

* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::visiblePositionForPoint const):
* Source/WebCore/rendering/HitTestRequest.h:

Add a new `HitTestSource` member to `HitTestRequest` that indicates whether the hit test request is
directly invoked through `Script`, or if it's user-triggered. We use this flag to prefer specified
`visibility()` over `usedVisibility()` (effectively ignoring all visibility adjustment).

(WebCore::HitTestRequest::assertConsistentType):
(WebCore::HitTestRequest::HitTestRequest):
(WebCore::HitTestRequest::userTriggered const):
* Source/WebCore/rendering/HitTestSource.h: Added.
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::positionForPointRespectingEditingBoundaries):
(WebCore::RenderBlock::positionForPointWithInlineChildren):
(WebCore::isChildHitTestCandidate):
(WebCore::RenderBlock::positionForPoint):
* Source/WebCore/rendering/RenderBlock.h:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::positionForPointWithInlineChildren):
(WebCore::RenderBlockFlow::positionForPoint):
* Source/WebCore/rendering/RenderBlockFlow.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::positionForPoint):
* Source/WebCore/rendering/RenderBox.h:
* Source/WebCore/rendering/RenderElementInlines.h:
(WebCore::RenderElement::visibleToHitTesting const):
* Source/WebCore/rendering/RenderFileUploadControl.cpp:
(WebCore::RenderFileUploadControl::positionForPoint):
* Source/WebCore/rendering/RenderFileUploadControl.h:
* Source/WebCore/rendering/RenderFragmentContainer.cpp:
(WebCore::RenderFragmentContainer::positionForPoint):
* Source/WebCore/rendering/RenderFragmentContainer.h:
* Source/WebCore/rendering/RenderInline.cpp:
(WebCore::RenderInline::positionForPoint):
* Source/WebCore/rendering/RenderInline.h:
* Source/WebCore/rendering/RenderLineBreak.cpp:
(WebCore::RenderLineBreak::positionForPoint):
* Source/WebCore/rendering/RenderLineBreak.h:
* Source/WebCore/rendering/RenderMultiColumnSet.cpp:
(WebCore::RenderMultiColumnSet::positionForPoint):
* Source/WebCore/rendering/RenderMultiColumnSet.h:
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::positionForPoint):
* Source/WebCore/rendering/RenderObject.h:

Plumb this new `HitTestRequest` flag through `RenderObject::positionForPoint()`, and propagate it
through all call sites and subclass implementations, recursively.

* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::RenderReplaced::paint):
* Source/WebCore/rendering/RenderReplaced.h:
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::positionForPoint):
* Source/WebCore/rendering/RenderText.h:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::isVisibleToHitTesting):
* Source/WebCore/rendering/svg/RenderSVGImage.cpp:
(WebCore::RenderSVGImage::nodeAtPoint):
* Source/WebCore/rendering/svg/RenderSVGInlineText.cpp:
(WebCore::RenderSVGInlineText::positionForPoint):
* Source/WebCore/rendering/svg/RenderSVGInlineText.h:
* Source/WebCore/rendering/svg/RenderSVGShape.cpp:
(WebCore::RenderSVGShape::nodeAtPoint):
* Source/WebCore/rendering/svg/RenderSVGText.cpp:
(WebCore::RenderSVGText::nodeAtFloatPoint):
(WebCore::RenderSVGText::nodeAtPoint):
(WebCore::RenderSVGText::positionForPoint):
* Source/WebCore/rendering/svg/RenderSVGText.h:
* Source/WebCore/rendering/svg/SVGInlineTextBox.cpp:
(WebCore::SVGInlineTextBox::nodeAtPoint):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGImage.cpp:
(WebCore::LegacyRenderSVGImage::nodeAtFloatPoint):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGShape.cpp:
(WebCore::LegacyRenderSVGShape::nodeAtFloatPoint):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::computeElementLayout):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDocumentGtk.cpp:
(webkit_dom_document_element_from_point):
(webkit_dom_document_caret_range_from_point):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::selectionPositionInformation):
* Source/WebKitLegacy/mac/DOM/DOMDocument.mm:
(-[DOMDocument elementFromPoint:y:]):
(-[DOMDocument caretRangeFromPoint:y:]):

Canonical link: https://commits.webkit.org/276762@main
